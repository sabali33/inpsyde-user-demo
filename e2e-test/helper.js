import { setBrowserViewport } from "@wordpress/e2e-test-utils";
export const createInpsydeUser = async (data) => {
	await page.type("#first-name", "Eliasu");
	await page.type("#last-name", "Abraman");
	await page.type("#position", "Developer");
	await page.type("#intro", "I like myself");
	await page.type("#inpsyde-userfacebook", "http://fb.com/tester");
	await page.type("#inpsyde-userxing", "http://xing.com/tester");
	await page.type("#inpsyde-userlinkedin", "http://lkd.com/tester");
	await page.type("#inpsyde-usergithub", "http://gt.com/tester");
	// await page.click(".interface-pinned-items");
	// await page.waitForSelector(".interface-interface-skeleton__sidebar");
	const arrows = await page.$$(
		".components-button.components-panel__body-toggle[aria-expanded='false']"
	);
	await Promise.all(arrows.map(async (arrow) => await arrow.click()));
	await page.waitForSelector(".editor-post-featured-image");
	await page.click(".editor-post-featured-image__toggle");
	await page.waitForSelector(".supports-drag-drop");
	await page.click(".save-ready");
	await page.click(".media-button-select");
};
