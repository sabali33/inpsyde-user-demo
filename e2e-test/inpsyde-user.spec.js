"use strict";

import {
	insertBlock,
	getEditedPostContent,
	createNewPost,
	enablePageDialogAccept,
	publishPost,
} from "@wordpress/e2e-test-utils";
import { createInpsydeUser } from "./helper";

describe("Insyde User", () => {
	//file path for plugin
	const plugin = "inpsyde-user-plugin";
	beforeAll(async () => {
		await enablePageDialogAccept();
	});

	it("Can add block", async () => {
		await createNewPost({ postType: "page", title: "New User" });
		await insertBlock("User Overview");
		expect(await page.$('[data-type="eliasu/inpsyde-user"]')).not.toBeNull();
		expect(await getEditedPostContent()).toMatchSnapshot();
		//await publishPost();
	});
	it("Can create Inpsyde user", async () => {
		await createNewPost({ postType: "inpsyde-user", title: "Eliasu Abraman" });
		await createInpsydeUser();
		//await publishPost();
	});
	it("Can set block attributes", async () => {
		await createNewPost({ postType: "page", title: "New User" });
		await insertBlock("User Overview");
		await page.type("#search-inpsyde-user-box > div", "Eliasu Abraman");
		await page.waitForSelector(
			".components-autocomplete__result.components-button"
		);
		await page.click(".components-autocomplete__result.components-button");
		expect(await getEditedPostContent()).toMatchSnapshot();
	});
});
