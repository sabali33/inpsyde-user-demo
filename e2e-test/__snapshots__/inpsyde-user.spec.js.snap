// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insyde User Can add block 1`] = `
"<!-- wp:eliasu/inpsyde-user -->
<div class=\\"wp-block-eliasu-inpsyde-user\\"><div class=\\"user-card\\"><div class=\\"full-name\\"><span class=\\"label\\">Name:</span><span class=\\"first-name\\"></span><span class=\\"last-name\\"></span></div><div class=\\"position\\"><span class=\\"label\\">Position:</span><span></span></div></div><div class=\\"modal-wrap active\\"><button class=\\"close-modal button-danger\\">Close</button><div class=\\"modal-content\\"><div class=\\"full-name\\"><span class=\\"first-name\\"></span><span class=\\"last-name\\"></span></div><div class=\\"intro\\"></div><ul class=\\"social-media-networks\\"></ul></div></div></div>
<!-- /wp:eliasu/inpsyde-user -->"
`;

exports[`Insyde User Can set block attributes 1`] = `
"<!-- wp:eliasu/inpsyde-user -->
<div class=\\"wp-block-eliasu-inpsyde-user\\"><div class=\\"user-card\\"><img src=\\"http://localhost:3601/wp-content/uploads/2021/11/264C30B3DE367D61C89AE0C9C087D87C.jpg\\" alt=\\"Eliasu\\"/><div class=\\"full-name\\"><span class=\\"label\\">Name:</span><span class=\\"first-name\\">Eliasu</span><span class=\\"last-name\\">Abraman</span></div><div class=\\"position\\"><span class=\\"label\\">Position:</span><span>Developer</span></div></div><div class=\\"modal-wrap active\\"><button class=\\"close-modal button-danger\\">Close</button><div class=\\"modal-content\\"><div class=\\"full-name\\"><span class=\\"first-name\\">Eliasu</span><span class=\\"last-name\\">Abraman</span></div><div class=\\"intro\\">I like myself</div><ul class=\\"social-media-networks\\"><li class=\\"facebook\\"><a href=\\"http://fb.com/tester\\">Facebook</a></li><li class=\\"github\\"><a href=\\"http://gt.com/tester\\">Github</a></li><li class=\\"xing\\"><a href=\\"http://xing.com/tester\\">Xing</a></li><li class=\\"linkedin\\"><a href=\\"http://lkd.com/tester\\">LinkedIn</a> </li></ul></div></div></div>
<!-- /wp:eliasu/inpsyde-user -->"
`;
